src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -44,15 +44,20 @@ public function __construct(UrlGeneratorInterface $urlGenerator, UserProviderInt
         ], $options);
     }
 
-    public function createLoginLink(UserInterface $user, Request $request = null, int $lifetime = null): LoginLinkDetails
+    /**
+     * @param array<string, \Stringable|scalar> $parameters A list of additional query string parameters that should be part of the login link
+     */
+    public function createLoginLink(UserInterface $user, Request $request = null, int $lifetime = null, array $parameters = []): LoginLinkDetails
     {
         $expires = time() + ($lifetime ?: $this->options['lifetime']);
         $expiresAt = new \DateTimeImmutable('@'.$expires);
 
-        $parameters = [
+        $hash = $this->signatureHasher->computeSignatureHash($user, $expires, $parameters);
+
+        $parameters += [
```

Comment:
⚠️  Warning! 

By using` += ` there, the priority is given to `$parameters`  (the first array) 

The potential shared keys won't be erased... so $user, $hash or $expires could be predefined by values in $parameters and not overwritten.

```php

function  createLoginLink(array $parameters) {
	$hash = '$this->signatureHasher->computeSignatureHash($user, $expires, $parameters)';
	$parameters += [
	    'user' => '$user->getUserIdentifier()',
	    'expires' => '$expires',
	    'hash' => $hash,
	];
	var_dump($parameters);
}

createLoginLink(['user' => 'not the one', 'hash' => 'a pre computed hash']);
```

```php
array(3) {
  ["user"]=>
  string(11) "not the one"
  ["hash"]=>
  string(19) "a pre computed hash"
  ["expires"]=>
  string(8) "$expires"
}
```

Reply:
Good point, I'll swap for `array_merge`.